presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -407,8 +407,17 @@ public <T extends PrestoSparkTaskOutput> IPrestoSparkTaskExecutor<T> doCreate(
         TaskId taskId = new TaskId(new StageExecutionId(stageId, 0), partitionId);
 
         // TODO: Remove this once we can display the plan on Spark UI.
-
-        log.info(PlanPrinter.textPlanFragment(fragment, functionAndTypeManager, session, true));
+        // Currently, `textPlanFragment` throws an exception if json-based UDFs are used in the query, which can only
+        // happen in native execution mode. To resolve this error, `JsonFileBasedFunctionNamespaceManager` muse be
+        // loaded on the executors as well (which is actually not required for native execution). To do so, we need a
+        // mechanism to ship the JSON file containing the UDF metadata to workers, which does not exist as of today.
+        // TODO - Add support for loading `JsonFileBasedFunctionNamespaceManager` on workers.
```

Comment:
can you create a github issue and link that here.

Reply:
Sure. I'll update the comment w/ the issue: https://github.com/prestodb/presto/issues/19600